standardization/src/main/scala/za/co/absa/enceladus/standardization/interpreter/stages/TypeParser.scala

```diff
@@ -36,65 +35,122 @@ object TypeParser {
-          if (format.isDefault) {
-            throw new InvalidParameterException("Dates & times represented as numeric values need specified 'pattern' metadata")
-          }
-          val Array(l, r) = format.split("\\.")
-          val precision = l.length() + r.length()
-          val scale = r.length()
-          attr.cast(DecimalType.apply(precision, scale)).cast(StringType)
-        case _: DecimalType =>
-          logger.warn(s"$path is specified as timestamp or date, but original type is Decimal. Trying to interpret as string.")
-          if (format.isDefault) {
-            throw new InvalidParameterException("Dates & times represented as numeric values need specified 'pattern' metadata")
-          }
-          attr.cast(StringType)
-        case _ => attr
-      }
-      castFunction(sourceCol, format)
+  private def dateTimePreProcess(path: String,
+                                 attr: Column,
+                                 pattern: DateTimePattern,
+                                 origType: DataType,
+                                 castFunction: colStrToColFnc): Column = {
+    val sourceCol = origType match {
+      case _: DoubleType | _: FloatType =>
+        logger.warn(s"$path is specified as timestamp or date, but original type is ${origType.typeName}. " +
+          "Trying to interpret as string.")
+        if (pattern.isDefault) {
+          throw new InvalidParameterException(
+            s"Dates & times represented as ${origType.typeName} values need specified 'pattern' metadata"
+          )
+        }
+        val index = pattern.indexOf(".") //This can stop working when Spark becomes Locale aware
+        val (precision, scale) = if (index == -1) {
+          (pattern.length, 0)
+        } else {
+          (pattern.length-1, pattern.length - index - 1)
+        }
+        attr.cast(DecimalType.apply(precision, scale)).cast(StringType)
+      case _: StringType => attr
+      case _ =>
+        logger.warn(s"$path is specified as timestamp or date, but original type is ${origType.typeName}. " +
+          "Trying to interpret as string.")
+        if (pattern.isDefault) {
+          throw new InvalidParameterException(
+            s"Dates & times represented as ${origType.typeName} values need specified 'pattern' metadata"
+          )
+        }
+        attr.cast(StringType)
     }
+    castFunction(sourceCol, pattern)
   }
 
+  private def castEpoch(attr: Column, epochFactor: Long, epochPattern: String, castFunction: colStrToColFnc): Column =
+    castFunction(from_unixtime(attr.cast("Long")  / epochFactor, epochPattern), epochPattern)
+
+  private def toTimestampWithDefaultTimeZone(col: Column, pattern: String, defaultTimeZone: String): Column =
+    to_utc_timestamp(to_timestamp(col, pattern), defaultTimeZone)
+
+  private def toDateWithDefaultTimeZone(col: Column, pattern: String, defaultTimeZone: String): Column =
```

Comment:
The name `toDateWithDefaultTimeZone` and parameter `defaultTimeZone` would make people think they can alter the output of date with timezone.